### PR TITLE
feat: use auth volume for dynamic reloads; clarify AwsAuth struct

### PR DESCRIPTION
**Pull request by TylerGillson**
This PR introduces support for [dynamic reloading](https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets-as-files-from-a-pod) of AWS credentials from a volume. Whereas environment variables populated from a secret are not reloaded by K8s until the pod is bounced.

Other:
- Unused fields were removed from the `AwsAuth` struct for clarity
- General documentation improvements in `values.yaml`
- Mandatory fields in `values.yaml` are now validated via Helm's `required` function

### CODE REVIEW

File: hack/extra-values.yaml [hunk 1]
```diff
@@ -1,5 +1,14 @@
...
+    # accessKeyId: ""
+    # secretAccessKey: ""
+    # sessionToken: ""
+    # By default, a secret will be created. Leave the above fields blank and specify 'createSecret: false' to use an existing secret.
+    # WARNING: the existing secret must match the format used in auth-secret.yaml
+    # createSecret: true
+
+  # Override the service account used by AWS validator (optional, could be used for IMDSv2 on EKS)
+  # WARNING: the chosen service account must have the same RBAC privileges as seen in manager-rbac.yaml
+  serviceAccountName: ""
```
Comment (ahmad-ibra) [inline hack/extra-values.yaml:14]: whats the reason for having this `extra-values.yaml`? it looks like all the values that can be defined here could also be defined in the more common `values.yaml`

File: chart/validator-plugin-aws/templates/deployment.yaml [hunk 2]
```diff
@@ -68,5 +71,15 @@ spec:
...
       serviceAccountName: {{ include "chart.fullname" . }}-controller-manager
-      terminationGracePeriodSeconds: 10
\ No newline at end of file
+      {{- end }}
+      terminationGracePeriodSeconds: 10
+      {{- if .Values.auth.secret }}
+      volumes:
+      - name: aws-auth
+        secret:
+          secretName: {{ .Values.auth.secret.secretName }}
```
Comment (ahmad-ibra) [inline chart/validator-plugin-aws/templates/deployment.yaml:84]: its possible that no `secretName` isnt defined when we enter this if block. is there any issue with that? ie should the if condition on line 80 be something like this?

```
{{- if and .Values.auth.secret .Values.auth.secret.secretName }}
```

**Review by ahmad-ibra — APPROVED** [review]
Left a few questions but nothing blocking, lgtm

File: chart/validator-plugin-aws/templates/deployment.yaml (hunk 2 above)
Comment (TylerGillson) [reply chart/validator-plugin-aws/templates/deployment.yaml:84]: Yeah, theoretically someone could leave `secret.secretName` empty, but there isn't much we can do to protect against that.

File: hack/extra-values.yaml (hunk 1 above)
Comment (TylerGillson) [reply hack/extra-values.yaml:14]: It's a holdover from when I was first autogenerating the chart. There's a make target that appends this file to the autogenerated values.yaml. I should probably just remove it now.

File: chart/validator-plugin-aws/templates/deployment.yaml (hunk 2 above)
Comment (TylerGillson) [reply chart/validator-plugin-aws/templates/deployment.yaml:84]: Actually, evidently I don't [know my template functions](https://helm.sh/docs/howto/charts_tips_and_tricks/#know-your-template-functions). Will fix this.

**Pull request merged**